h/oauth/schemas.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+
+import colander
+
+from h import i18n
+from h.models.auth_client import GrantType, ResponseType
+from h.schemas.base import CSRFSchema
+
+_ = i18n.TranslationString
+
+
+def enum_schema_node(enum_cls, **kwargs):
+    """
+    Return a `colander.SchemaNode` for a field with a given enum type.
+
+    :param enum_cls: The enum class
+    :type enum_cls: enum.Enum
+    """
+    values = []
+    for name, _ in enum_cls.__members__.items():
+        values.append(name)
+
+    return colander.SchemaNode(colander.String(),
+                               validator=colander.OneOf(values),
+                               **kwargs)
+
+
+class CreateAuthClientSchema(CSRFSchema):
```

Comment:
This PR is getting quite large so, as long as the constraints we want are enforced in the DB, I think it might be better if I add more user-friendly validation in a follow-up PR?